starter-core/src/main/java/io/micronaut/starter/feature/build/BuildCacheConfiguration.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.build;
+
+import io.micronaut.core.annotation.Nullable;
+
+public interface BuildCacheConfiguration {
```

Comment:
should not this be in the buildless package?

Reply:
@sdelamo this is a general configuration for any kind of Gradle build caching, following the structure of their [build cache configuration](https://docs.gradle.org/current/userguide/build_cache.html).

I figure if there are future providers, or some feature which needs to activate the build cache, it can use this configuration. The Buildless feature uses this configuration the same way. If you'd like I can move it to be next to the `Buildless` feature.

Reply:
I'm not sure we need this in this PR...

The work to support swappable build caches is probably out of scope here

Unless I'm missing why this is here

Reply:
Agreed, removed (easy to add in the future)